src/nanoarrow/device/metal.cc

```diff
@@ -292,7 +297,7 @@ static int ArrowDeviceMetalCopyRequiredCpuToMetal(MTL::Device* mtl_device,
 }
 
 static ArrowErrorCode ArrowDeviceMetalSynchronize(struct ArrowDevice* device,
-                                                  void* sync_event,
+                                                  void* sync_event, void* stream,
```

Comment:
should we return a different status than `NANOARROW_OK` since it's not implemented?